### PR TITLE
1. Draft rewrite

### PR DESCRIPTION
**Pull request by spirillen**
As @funilrys was a little to fast for me to re-read my suggestions in this [merge](https://github.com/funilrys/PyFunceble/commit/26328378b0c20bb18cad65b4a96a882a3f82379f) , I'll submit some better here :smile: :laughing:

### CODE REVIEW

File: version.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-current_version: 3.0.0.dev (Teal Blauwbok)
+current_version: 3.0.1.dev (Teal Blauwbok)
```
Comment (funilrys) [inline version.yaml]: @spirillen it's generated with `PyFunceble --prod` (`--production`)

Comment (funilrys) [reply version.yaml]: So it's important to work from an isolated environment 😉

Comment (spirillen) [reply version.yaml]: So you are asking me to undo changes in this particular file?

File: version.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-current_version: 3.0.1.dev (Teal Blauwbok)
+current_version: 3.0.0.dev (Teal Blauwbok)
```
Comment (funilrys) [inline version.yaml]: @spirillen Hehe you are not in an isolated environment :wink:

Comment (spirillen) [reply version.yaml]: Well haven't noticed this little trick of your... I was in this case looking at your tag/3.0.0 commit for version number changes :package:

**Comment by spirillen** [comment]
PS: @funilrys I still consider this a draft for revisit and hopefully some constructive feedback from others :de:

File: version.yaml (hunk 2 above)
Comment (funilrys) [reply version.yaml]: @spirillen 
```
$ cd /path/to/PyFunceble
$ virtualenv -p python3 venv
$ source venv/bin/activate
$ pip3 install -e .
[... Edit PyFunceble ...]
$ PyFunceble --prod
[... Do what you have to do ...] :wink: `

**Comment by funilrys** [comment]
We could also provide a multilingual tool :smile_cat:  But that's not in my plan yet!

File: version.yaml (hunk 2 above)
Comment (spirillen) [reply version.yaml]: Well I'll write me a note on that one :+1:

**Comment by spirillen** [comment]
> We could also provide a multilingual tool smile_cat But that's not in my plan yet!

Puuuhhh I think I'm busy enough rewriting/writing docs.....

File: version.yaml (hunk 2 above)
Comment (spirillen) [reply version.yaml]: Have changed this back

File: docs/components/custom-dns-server.rst [hunk 3]
```diff
@@ -26,8 +34,17 @@ to
...
+        - "9.9.9.9"
+        - "149.112.112.112"
+
+
+.. version:: 3.0.0
+
+Since :version:`3.0.0` it is possible to assign a specific port to use with the 
+DNS-Server.
+
+.. example::
```
Comment (funilrys) [inline docs/components/custom-dns-server.rst]: the `.. example::` Is not known by read the docs...  Please report to https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#admonitions

File: docs/facts/faq.rst [hunk 4]
```diff
@@ -21,8 +21,10 @@ I have multiple CPU
 
 Simply use the :code:`-m | --multiprocess` argument to activate
 the usage of multiple processes.
-You can in addition to that use the :code:`-p | --processes`
-argument to tell set the number of simultaneous process to run.
+You should in addition to the :code:`-m` specify the :code:`-p | --processes`
+argument as the default value is 25 simultaneous processes. This number will
```
Comment (funilrys) [inline docs/facts/faq.rst]: If `-p` is avoided, we follow the number of available CPU cores @spirillen (Documenting will fix #78)

File: docs/facts/faq.rst [hunk 5]
```diff
@@ -34,6 +36,17 @@ argument to tell set the number of simultaneous process to run.
     ::
 
         $ lscpu | grep -E '^Thread|^Core|^Socket|^CPU\('
+    
+    or
+    
+    ::
+	$ ``$(nproc --ignore=1)``
```
Comment (funilrys) [inline docs/facts/faq.rst]: Newline expected.

As example: the previous block.

File: docs/facts/faq.rst [hunk 6]
```diff
@@ -43,7 +56,7 @@ I do not have multiple CPU
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 In case you only have a single core, you should disable the usage of
-the WHOIS lookup by adding the :code:`-.no-whois` to your command line
+the WHOIS lookup by adding the :code:`-no-whois` to your command line
```
Comment (funilrys) [inline docs/facts/faq.rst]: it's `--no-whois`.

File: docs/usage/from-a-terminal.rst [hunk 7]
```diff
@@ -725,166 +791,274 @@ Global overview
                     [--commit-results-message COMMIT_RESULTS_MESSAGE] [--clean]
```
Comment (funilrys) [inline docs/usage/from-a-terminal.rst]: I will generate this section with a script I have locally :-)

**Comment by funilrys** [comment]
Note to self after merging: 
* Update global overview of the terminal usage.
* Update the `What can we do` section of the `README.rst`.

File: docs/facts/faq.rst (hunk 4 above)
Comment (spirillen) [reply docs/facts/faq.rst]: Have rewritten the sentence(s)

Have a second look :smile:

File: docs/components/custom-dns-server.rst [hunk 8]
```diff
@@ -4,8 +4,11 @@ Custom DNS Server
 Why do we need it?
 ------------------
 
-As some time we can't trust the default DNS server into a container/machine, we allow you to communicate us a list
-of DNS server to ask for the records.
+Some times the testing environment is setup to use DNS-server which isn't 
+suited for running a PyFunceble test of actually expired or active domains or
```
[the remark this answers is not in the document]
Comment (spirillen) [reply docs/components/custom-dns-server.rst]: I think the usage of `a` is the right one, as it references to `a | this` specified test. The next test could be of the default `DNS:PORT` 

But as always, I'm open for arguments

File: docs/components/custom-dns-server.rst (hunk 3 above)
Comment (spirillen) [reply docs/components/custom-dns-server.rst]: Based on the https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#admonitions The use of `hint` seems like a better chois of `wording` for this example box. Alternatively it should be converted into a [Code Blocks](https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#code-blocks)

```
.. code-block:: shell
    --dns 95.216.209.53:53 116.203.32.67:53 9.9.9.9:853
```

File: docs/usage/from-a-terminal.rst (hunk 7 above)
Comment (spirillen) [reply docs/usage/from-a-terminal.rst]: I'm a bit unsure about what you would like me to do here.....

File: docs/components/custom-dns-server.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (funilrys) [reply docs/components/custom-dns-server.rst]: All good :+1:

File: docs/usage/from-a-terminal.rst (hunk 7 above)
Comment (funilrys) [reply docs/usage/from-a-terminal.rst]: Nothing to do :smile_cat: I just wanted to inform you that that section will be overwritten with the changes made inside the code (what we get from `--help`).

**Pull request merged**